### ISSUE TITLE
DDLDatabase fails with MySQL ON UPDATE

### ISSUE THREAD

**Issue by anuraaga** [jOOQ/jOOQ#7019]
Parser does not seem to recognize ON UPDATE (e.g., on a timestamp).

```
org.jooq.impl.ParserException: C42000_NO_SUBCLASS: Expected tokens: [CHECK, PRIMARY KEY, UNIQUE]: d_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL [*]ON UPDATE CURRENT_TIMESTAMP,
  created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP NO
        at org.jooq.impl.ParserImpl$ParserContext.unexpectedToken(ParserImpl.java:5791)
        at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:547)
        at org.jooq.impl.ParserImpl.parse(ParserImpl.java:345)
        at org.jooq.impl.ParserImpl.parse(ParserImpl.java:337)
        at org.jooq.util.ddl.DDLDatabase.create0(DDLDatabase.java:124)
        at org.jooq.util.AbstractDatabase.create(AbstractDatabase.java:237)
        at org.jooq.util.AbstractDatabase.create(AbstractDatabase.java:226)
        at org.jooq.util.AbstractDatabase.getDialect(AbstractDatabase.java:208)
        at org.jooq.util.AbstractGenerator.logDatabaseParameters(AbstractGenerator.java:131)
        at org.jooq.util.JavaGenerator.generate(JavaGenerator.java:271)
        at org.jooq.util.GenerationTool.run(GenerationTool.java:642)
        at org.jooq.util.GenerationTool.generate(GenerationTool.java:203)
        at org.jooq.util.GenerationTool.main(GenerationTool.java:175)
```

**Comment by lukaseder** [comment]
That's a very interesting feature. I actually needed this the other day in Oracle. Of course, this can always be implemented through triggers, but the more declarative DDL-only approach looks nice. I've created an issue to support the clause in jOOQ, natively: #7045. For the time being, we'll simply parse and ignore the clause.

**Comment by lukaseder** [comment]
Fixed for 3.11 and will be merged to 3.10.4 (#7046)

**Comment by lukaseder** [comment]
Commit: https://github.com/jOOQ/jOOQ/commit/12bb3efbcc338ec7813ffcab793663f97c76780a
